### PR TITLE
Create 2021-09-27-solidity-0.8.8-release-announcement.md

### PR DESCRIPTION
**Pull request by franzihei**

### CODE REVIEW

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.8 Release Announcement'
+date: '2021-09-27'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.7](https://github.com/ethereum/solidity/releases/tag/v0.8.8) introduces .....
```
Comment (bshastry) [inline _posts/2021-09-27-solidity-0.8.8-release-announcement.md:10]:
```suggestion
[Solidity v0.8.8](https://github.com/ethereum/solidity/releases/tag/v0.8.8) introduces .....
```

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.8 Release Announcement'
+date: '2021-09-27'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.8](https://github.com/ethereum/solidity/releases/tag/v0.8.8) introduces .....
```
Comment (hrkrshnn) [inline _posts/2021-09-27-solidity-0.8.8-release-announcement.md:10]:
```suggestion
[Solidity v0.8.8](https://github.com/ethereum/solidity/releases/tag/v0.8.8) introduces user defined value types as a major feature. Details can be found in the [blog post](https://deploy-preview-148--solidity-blog.netlify.app/2021/09/13/user-defined-value-types/). 
<!-- TODO fix link -->
```

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md [hunk 3]
```diff
@@ -101,6 +101,31 @@ inside the constructor but you cannot use them to initialize other
 immutable variables at the point of their declaration.
 
 
+### Import Paths
+
+We added a new option `--include-path` to `solc` that hopefully makes it much easier to compile
+more complicated projects using multiple libraries. You can use it to specify
```
Comment (cameel) [inline _posts/2021-09-27-solidity-0.8.8-release-announcement.md:107]: Might be worth clarifying that it affects the default import callback. I.e. all CLI usage, with or without `--standard-json` but not usage via solc-js (unless someone uses the `solcjs` script to invoke the compiler).

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md [hunk 4]
```diff
@@ -101,6 +101,31 @@ inside the constructor but you cannot use them to initialize other
...
+directories where source files are looked up when `import` statements are resolved. If you use
+`npm` to install Solidity packages, they are usually placed in a subdirectory below
+`node_modules`. If you use imports of the form `import X from "@packagename/contracts/x.sol";`,
+then invoking the compiler as follows should properly resolve all imports without further work:
+
+```bash
+solc contract.sol --base-path . --include-path node_modules/
+```
```
Comment (cameel) [inline _posts/2021-09-27-solidity-0.8.8-release-announcement.md:115]: Do we want to mention that base path and include paths are automatically added to allowed paths? "without further work" kinda means that but it's not very explicit.

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+### User Defined Value Types
+
+A user defined value type allows creating a zero-cost-abstraction over an elementary value type that
+also increases type safety and improves readability. Details can be found in the focussed [blog
+post](https://deploy-preview-148--solidity-blog.netlify.app/2021/09/13/user-defined-value-types/).
+
+<!-- TODO fix link -->
+
+### Inheritance
```
Comment (franzihei) [inline _posts/2021-09-27-solidity-0.8.8-release-announcement.md:24]: Is that sub-title expressive enough? Maybe it's too generic?

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    return type(AuctionState).min;
+}
+
+// returns AuctionState.Completed
+function max() pure returns (AuctionState) {
+    return type(AuctionState).max;
+}
+```
+
+### Immutable Variables
```
Comment (franzihei) [inline _posts/2021-09-27-solidity-0.8.8-release-announcement.md:69]: Subtitle too generic?

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+As you probably know, it is also possible to initialize immutable variables
+at the point of their declaration, for example `uint immutable deployTime = block.timestamp;`.
+Because the order of initialization is not
+always obvious, such immutable variables are only considered initialized once
+the constructor is executed. This means you can read from such variables
+inside the constructor but you cannot use them to initialize other
+immutable variables at the point of their declaration.
+
+
+### Import Paths
```
Comment (franzihei) [inline _posts/2021-09-27-solidity-0.8.8-release-announcement.md:104]: Subtitle too generic?

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md (hunk 3 above)
Comment (chriseth) [reply _posts/2021-09-27-solidity-0.8.8-release-announcement.md:107]: I considered it, but isn't that use-case a bit too specific and might confuse the "ordinary peopel"? I mean this talk about "options"  - and if you use the binary interface, there is no real place to provide these options, isn't it?

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md (hunk 4 above)
Comment (chriseth) [reply _posts/2021-09-27-solidity-0.8.8-release-announcement.md:115]: yep, will do.

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md (hunk 3 above)
Comment (cameel) [reply _posts/2021-09-27-solidity-0.8.8-release-announcement.md:105]: True, but what I'm worried about is that it's not clear that it affects `--standard-json` too. Though maybe it's ok - most people probably are also not aware that with `--standard-json` imports might still get loaded from disk.

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md [hunk 8]
```diff
@@ -101,14 +99,15 @@ inside the constructor but you cannot use them to initialize other
 immutable variables at the point of their declaration.
 
 
-### Import Paths
+### Import Include Paths
 
 We added a new option `--include-path` to `solc` that hopefully makes it much easier to compile
 more complicated projects using multiple libraries. You can use it to specify
```
Comment (cameel) [inline _posts/2021-09-27-solidity-0.8.8-release-announcement.md]: Oh, one more thing. We might also say that `--include-path` is meant to be an alternative to import remapping in simple cases (i.e. where they are not used to rename the file itself). For example Brownie currently uses remapping to resolve dependencies and I think that they could switch to this feature instead.

Comment (chriseth) [reply _posts/2021-09-27-solidity-0.8.8-release-announcement.md]: Added another paragraph.

File: _posts/2021-09-27-solidity-0.8.8-release-announcement.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+```
+
+The previously introduced option `--base-path` provides a path where the lookup is performed
+first. If it fails, the include paths are searched for the file.
+
+It is often a source of frustration that the Solidity compiler stores the full path
+to a source file as part of the metadata, which sometimes makes it more difficult to
+reproduce a compilation when not using the standard-json mode.
+
+If you use `--base-path` and `--include-path`, these paths will not be stored in
```
Comment (cameel) [inline _posts/2021-09-27-solidity-0.8.8-release-announcement.md:124]: It reads a bit as if "paths" was referring to file paths rather than paths passed to these options. So someone might think that file paths are not stored in metadata at all in this case. I'd reword it to make it completely clear:
```suggestion
If you use `--base-path` and `--include-path`, then only relative paths to files located inside them will be stored in
```

**Comment by chriseth** [comment]
I'll squash.

**Review by chriseth — APPROVED** [review]
